www/nejj02/hw08/homework.js

```diff
@@ -0,0 +1,222 @@
+ * console.log('Ahoj světe');
+ */
+
+console.log('Hello, World!');
+
+/**
+ * 1) Pepe's age. Vypište na konzoli smysluplnou oznamovací větu s věkem Pepy, pokud znáte jeho rok narození, 
+ * který je uložený v proměnné a pro výpis použijte zřetězení stringů nebo interpolaci. Pro názvy proměnných 
+ * používejte smysluplnou angličtinu.
+ */
+// Solution here
+
+let birth = 2000;
+let age = new Date().getFullYear() - birth;
+console.log('Pepe is or will be ' + age + ' years old this year!')
+
+/**
+ * 2) WTF (wow, that's fun). Vypište na konzoli teplotu v Fahrenheiht, pokud znáte teplotu v Celsius, a také naopak. 
+ * Formát výpisu je: 20°C =  68°F resp. 68°F = 20°C. Opět používejte proměnné. Výpočet probíhá takto:
+ *     z C na F: vynásobit 9, vydělit 5 a přičíst 32. 
+ *     z F na C: odečíst 32, vynásobit 5 a vydělit 9. 
+ */
+// Solution here
+
+let temperatureInCelsius = 30;
+let temperatureInFahrenheit = 451;
+
+let resultInFahrenheit = temperatureInCelsius * 9 / 5 + 32;
+let resultInCelsius = (temperatureInFahrenheit - 32) * 5 / 9
+
+console.log(temperatureInCelsius + '°C = ' + resultInFahrenheit + '°F')
+console.log(temperatureInFahrenheit + '°F = ' + resultInCelsius + '°C')
+
+/**
+ * 3) Funkce function fonction funktio. Vemte předchozí úlohy a udělejte z nich funkce. Tj. vytvoříte funkce, 
+ * které přijímají argumenty, a na základě argumentů po zavolání vypíše výsledek na konzoli. 
+ * Párkrát zavolejte tyto funkce s různými argumenty. V konzoli také vyzkoušejte, zda fungují vaše funkce. 
+ * 
+ * Pro testování funkce:
+ * - Pouze pomocí JavaScriptu (bez knihoven) vytvořte HTML tlačítko s názvem této úlohy, resp. co funkce dělá, a 
+ * id s číslem úlohy <button id="task-1">Uloha 1 (Pepe's age)</button>, umístěte ho na stránku do předem vytvořeného 
+ * místa <div id="tasks"></div> a pomocí posluchače události "click" nabindujte implementovanou funkci na toto tlačítko.
+ * 
+ * Výsledkem má být tlačítko, na které když kliknete, tak se provede to, co je implementováno ve funkci.
+ * 
+ */
+// Solution here
+
+const hello = () => {
+    return 'Hello, World!';
+}
+
+const printAge = (birth) => {
+    let age = new Date().getFullYear() - birth;
+    return 'Pepe is or will be ' + age + ' years old this year!';
+}
+
+const toFahrenheit = (celsius) => {
+    let result = celsius * 9 / 5 + 32;
+    return celsius + '°C = ' + result + '°F'
+}
+
+const toCelsius = (fahrenheit) => {
+    let result = fahrenheit * 9 / 5 + 32;
+    return fahrenheit + '°F = ' + result + '°C';
+}
+
+const createButton = (id, text, action) => {
+    const button = document.createElement('button');
+    button.id = id;
+    button.innerText = text;
+    button.addEventListener('click', () => {
+        document.getElementById('result').innerText = action;
+    })
+    document.getElementById('task-buttons').appendChild(button);
+}
+
+createButton('task-0', 'Task 0 (Pre-preparacion)', hello());

```

Comment:
to `hello()` neni spravne, funkce createButton by mela bindovat funkci (callback), ne navratovou hodnotu z volani funkce, protoze mate pocitat az ve chvili kdy uzivatel klikne na tlacitko
```js
const result = document.getElementById('result');
createButton('...', '...', () => {
  const value = printAge(2000);
  result.innterText = value;
});
```